### ISSUE TITLE
got an error while trying to load a polymorphic association

### ISSUE THREAD

**Issue by luizkowalski** [salsify/goldiloader#105]
Hi

I have the following structure:

```ruby
class User < ApplicationRecord
  ....
  has_many :app_notifications, inverse_of: :user, dependent: :destroy
end

class AppNotification < ApplicationRecord
  has_many :references,
           inverse_of: :app_notification,
           class_name: 'AppNotificationReference',
           dependent: :destroy
end

class AppNotificationReference < ApplicationRecord
  acts_as_paranoid

  belongs_to :app_notification, required: true, inverse_of: :references
  belongs_to :notifiable, -> { with_deleted }, polymorphic: true, required: true
end
```
I don't have any other validations.

after adding `goldiloader` i started seeing this error on my specs:

```
ArgumentError:
       Polymorphic associations do not support computing the class.
     # /usr/local/bundle/gems/goldiloader-4.0.0/lib/goldiloader/active_record_patches.rb:82:in `eager_loadable?'
     # /usr/local/bundle/gems/goldiloader-4.0.0/lib/goldiloader/active_record_patches.rb:114:in `eager_loadable?'
     # /usr/local/bundle/gems/goldiloader-4.0.0/lib/goldiloader/active_record_patches.rb:104:in `auto_include?'
     # /usr/local/bundle/gems/goldiloader-4.0.0/lib/goldiloader/active_record_patches.rb:121:in `load_with_auto_include'
     # /usr/local/bundle/gems/goldiloader-4.0.0/lib/goldiloader/active_record_patches.rb:143:in `find_target'
     # ./spec/domains/payments/card/services/authorization_handler_spec.rb:252:in `block (6 levels) in <top (required)>'


expect(app_notification.references.first.notifiable).to eq(payment_card)
````

I tried to add `-> { auto_include(false) }` to disable eager loading to the associations on app_notification, even on user. nothing helped. 

Is this an issue or misconfiguration?

**Comment by jturkel** [comment]
Thanks for reporting this issue. I'm able to reproduce the problem which occurs whenever using a polymorphic association with a scope block in Rails 6.0+. I'll work on getting a fix out this week.

**Comment by jturkel** [comment]
Fix released in 4.0.1. Thanks for reporting the issue @luizkowalski.
